.openshift-ci/scripts/lib.sh

```diff
@@ -140,3 +140,94 @@ get_pr_details() {
     _PR_DETAILS="$pr_details"
     echo "$pr_details"
 }
+
+import_creds() {
+    shopt -s nullglob
+    for cred in /tmp/secret/**/[A-Z]*; do
+        export "$(basename "$cred")"="$(cat "$cred")"
+    done
+}
+
+# Ensures that the given secret exists in the environment.
+# As a side effect, if the secret does not currently exist,
+# all secrets are imported (via import_creds) and then checked again.
+ensure_secret_exists() {
+    local secret_name="$1"
+
+    if [[ -z "${secret_name}" ]]; then
+        import_creds
+        if [[ -z "${secret_name}" ]]; then
+            die "No such secret called ${secret_name}"
+        fi
+    fi
+}
+
+copy_secret_to_file() {
+    local secret_name="$1"
+    shift
+    local destination="$1"
+    shift
+    local permissions="$1"
+    shift
+
+    ensure_secret_exists "$secret_name"
+
+    echo "${secret_name}" > "${destination}"
+    chmod "${permissions}" "${destination}"
+}
+
+get_secret_content() {
+    local secret_name="$1"
+
+    ensure_secret_exists "$secret_name"
+    echo "${secret_name}"
```

Comment:
Wait... How does this work? Lets assume we call `get_secret_content SOME_CREDENTIAL`, then line 182 would expand `secret_name` and call `ensure_secret_exists "SOME_CREDENTIAL"` the you return from the call and... `secret_name` is now magically the secret itself rather than the key? What?

Reply:
good spot... not sure what I was thinking here